stac_fastapi/api/tests/test_models.py

```diff
@@ -0,0 +1,100 @@
+import json
+
+import pytest
+from pydantic import ValidationError
+
+from stac_fastapi.api.models import create_get_request_model, create_post_request_model
+from stac_fastapi.extensions.core.filter.filter import FilterExtension
+from stac_fastapi.extensions.core.sort.sort import SortExtension
+from stac_fastapi.types.search import BaseSearchGetRequest, BaseSearchPostRequest
+
+
+def test_create_get_request_model():
+    extensions = [FilterExtension()]
+    request_model = create_get_request_model(extensions, BaseSearchGetRequest)
+
+    model = request_model(
+        collections="test1,test2",
+        ids="test1,test2",
+        bbox="0,0,1,1",
+        intersects=json.dumps(
+            {
+                "type": "Polygon",
+                "coordinates": [[[0, 0], [0, 1], [1, 1], [1, 0], [0, 0]]],
+            }
+        ),
+        datetime="2020-01-01T00:00:00Z",
+        limit=10,
+        filter="test==test",
+        # FIXME: hyphen aliases are not properly working
+        # **{"filter-crs": "epsg:4326", "filter-lang": "cql2-text"},
```

Comment:
do we know if this was working before pydantic 2.0 ?

Reply:
maybe this is linked to the changes at https://github.com/stac-utils/stac-fastapi/pull/625/files#diff-f84457de825c3fa96fa65673a6f85442ac57bd10f9f939cb7411780d3b6fabb5R47

Reply:
hmm, let me test that

Reply:
I tested locally and reverting back to using the fastapi.Body instead of FieldInfo doesn't change that.
This section only handles POST requests that use the Pydantic BaseModel.

The issue with aliases only appears with GET requests that use attrs classes with Pydantic Fields. I don't think that ever worked. I will do some more testing tomorrow.

Reply:
I did some more testing and I am confident that this bug already existed before my changes and I only discovered it. I added several tests and linked them back to the issue I created for the bug.